test/suite/test_chunkcache04.py

```diff
@@ -0,0 +1,141 @@
+            'chunk_cache=[enabled=true,chunk_size=512KB,capacity=20GB,pinned=' \
+                + '("' + '{}'.format("\",\"".join(self.pinned_uris)) \
+                + '"),type={},storage_path=WiredTigerChunkCache]'.format(self.chunk_cache_type)
+
+    def conn_extensions(self, extlist):
+        if os.name == 'nt':
+            extlist.skip_if_missing = True
+        extlist.extension('storage_sources', 'dir_store')
+
+    def get_stat(self, stat):
+        stat_cursor = self.session.open_cursor('statistics:')
+        val = stat_cursor[stat][2]
+        stat_cursor.close()
+        return val
+
+    def insert(self, uri, ds):
+        cursor = self.session.open_cursor(uri)
+        for i in range(1, self.rows):
+            cursor[ds.key(i)] = str(i) * 100
+    
+    def read_and_verify(self, uri, ds):
+        cursor = self.session.open_cursor(uri)
+        for i in range(1, self.rows):
+            cursor.set_key(ds.key(i))
+            cursor.search()
+            self.assertEqual(cursor.get_value(), str(i) * 100)
+
+    def test_chunkcache04(self):
+        uris = ["table:chunkcache03", "table:chunkcache04"]
+        ds = [SimpleDataSet(self, uri, 0, key_format=self.key_format, value_format=self.value_format) for uri in uris]
+
+        # Insert unpinned data into two tables. 
+        for i, dataset in enumerate(ds):
+            dataset.populate()
+            self.insert(uris[i], dataset)
+
+        # As we have not flushed yet, assert we have no newly inserted chunks. 
+        self.assertEqual(self.get_stat(wiredtiger.stat.conn.chunk_cache_newly_inserted), 0)
+        
+        self.session.checkpoint()
+        self.session.checkpoint('flush_tier=(enabled)')
+
+        # Assert that chunks are not pinned. 
+        self.assertEqual(self.get_stat(wiredtiger.stat.conn.chunk_cache_chunks_pinned), 0)
+
+        # Assert the new chunks are ingested. 
+        unpinned_ingest = self.get_stat(wiredtiger.stat.conn.chunk_cache_newly_inserted)
+        self.assertGreater(unpinned_ingest, 0)
+        
+        ds = [SimpleDataSet(self, uri, 0, key_format=self.key_format, value_format=self.value_format) for uri in self.pinned_uris]
+
+        # Insert pinned data into two tables.
+        for i, dataset in enumerate(ds):
+            dataset.populate()
+            self.insert(uris[i], dataset)
+
+        self.session.checkpoint()
+        self.session.checkpoint('flush_tier=(enabled)')
+
+        # Assert the new chunks are ingested.
+        pinned_ingest = self.get_stat(wiredtiger.stat.conn.chunk_cache_newly_inserted)
+        self.assertGreater(pinned_ingest, unpinned_ingest)
+
+        # Assert that chunks are pinned.
+        old_pinned = self.get_stat(wiredtiger.stat.conn.chunk_cache_chunks_pinned)
+        self.assertGreater(old_pinned, 0)
+
+        # Flush the pinned chunks
+        self.session.checkpoint()
+        self.session.checkpoint('flush_tier=(enabled)')
```

Comment:
Do we create new versions of objects every time we flush? I thought we would only do that if some work was done on that particular uri.

Reply:
you are correct - updated here [d63f634](https://github.com/wiredtiger/wiredtiger/pull/9752/commits/d63f634fe084eaeba7fbbce337e874452c45dd5f)